### ISSUE TITLE
consistent-destructuring

### ISSUE THREAD

**Issue by hornta** [eslint/eslint#11216]
**Please describe what the rule should do:**
Should report on usage of object properties in dot notation on an already destrucuted object.

**What category of rule is this? (place an "X" next to just one item)**

[ ] Warns about a potential error (problem)
[ ] Suggests an alternate way of doing something (suggestion)
[X] Enforces code style (layout)
[ ] Other (please specify:)

**Provide 2-3 code examples that this rule will warn about:**

```js
// 👎
const { foo } = bar;

bar.xyz() // `bar` has already been destructured.
```

```js
// 👍
const { foo, xyz } = bar;

xyz();
```

```js
// 👍 
const bar = { xyz: () => {} };

bar.xyz(); // allowed because bar has not previously been destructured
```

**Why should this rule be included in ESLint (instead of a plugin)?**
Seems reasonable. There is already a consistent-return rule.

**Are you willing to submit a pull request to implement this rule?**
Not right now because I do not have the time, but later yes
